backend/src/graphql/resolver/util/creations.test.ts

```diff
@@ -0,0 +1,268 @@
+          creationDate: contributionDateFormatter(now),
+        },
+      })
+      await mutate({
+        mutation: createContribution,
+        variables: {
+          amount: 500.0,
+          memo: 'Contribution for the last month',
+          creationDate: contributionDateFormatter(
+            new Date(now.getFullYear(), now.getMonth() - 1, now.getDate()),
+          ),
+        },
+      })
+    })
+
+    it('has the correct data setup', async () => {
+      await expect(Contribution.find()).resolves.toEqual([
+        expect.objectContaining({
+          userId: user.id,
+          contributionDate: setZeroHours(now),
+          amount: expect.decimalEqual(250),
+          memo: 'Admin contribution for this month',
+          moderatorId: admin.id,
+          contributionType: 'ADMIN',
+          contributionStatus: 'PENDING',
+        }),
+        expect.objectContaining({
+          userId: user.id,
+          contributionDate: setZeroHours(
+            new Date(now.getFullYear(), now.getMonth() - 1, now.getDate()),
+          ),
+          amount: expect.decimalEqual(160),
+          memo: 'Admin contribution for the last month',
+          moderatorId: admin.id,
+          contributionType: 'ADMIN',
+          contributionStatus: 'PENDING',
+        }),
+        expect.objectContaining({
+          userId: user.id,
+          contributionDate: setZeroHours(
+            new Date(now.getFullYear(), now.getMonth() - 2, now.getDate()),
+          ),
+          amount: expect.decimalEqual(450),
+          memo: 'Admin contribution for two months ago',
+          moderatorId: admin.id,
+          contributionType: 'ADMIN',
+          contributionStatus: 'PENDING',
+        }),
+        expect.objectContaining({
+          userId: user.id,
+          contributionDate: setZeroHours(now),
+          amount: expect.decimalEqual(400),
+          memo: 'Contribution for this month',
+          moderatorId: null,
+          contributionType: 'USER',
+          contributionStatus: 'PENDING',
+        }),
+        expect.objectContaining({
+          userId: user.id,
+          contributionDate: setZeroHours(
+            new Date(now.getFullYear(), now.getMonth() - 1, now.getDate()),
+          ),
+          amount: expect.decimalEqual(500),
+          memo: 'Contribution for the last month',
+          moderatorId: null,
+          contributionType: 'USER',
+          contributionStatus: 'PENDING',
+        }),
+      ])
+    })
+
+    describe('call getUserCreation now', () => {
+      it('returns the expected open contributions', async () => {
+        await expect(getUserCreation(user.id, 0)).resolves.toEqual([
+          expect.decimalEqual(550),
+          expect.decimalEqual(340),
+          expect.decimalEqual(350),
+        ])
+      })
+
+      describe('run forward in time one hour before next month', () => {
+        const targetDate = new Date(now.getFullYear(), now.getMonth() + 1, 0, 23, 0, 0)
```

Comment:
initializing the targetDate with now.month + 1 will set the next month, right?
But then the test in line 191 seems not correct, because it returns the same creation-amounts than in the test of line 161, where the time is in the current month.

Reply:
Javascript can be a mess. I really recommend to use the browser console, to see what is happening.
```
const date = new Date()
-> undefined
date
-> Date Tue Nov 15 2022 20:31:31 GMT+0100 (Central European Standard Time)
date.setDate(0)
-> 1667244691495
date
Date Mon Oct 31 2022 20:31:31 GMT+0100 (Central European Standard Time)
```
Doe that answer your question?